### PR TITLE
Supabase guide

### PR DESCRIPTION
**Pull request by thisisgoldman**

### CODE REVIEW

**Comment by thisisgoldman** [comment]
* **#203** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/203?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @thisisgoldman and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/guides/supabase.mdx [hunk 1]
```diff
@@ -1,7 +1,258 @@
...
+    ```sql
+    create publication sequin_pub for table table1, table2, table3;
+    ```
+  </Step>
+  <Step title="Enter the replication slot details in Sequin">
+    Back in the Sequin Console, enter the name of the replication slot (e.g. `sequin_slot`) and publication (e.g. `sequin_pub`) you just created. Then, name your database and click **Create Database**.
+  </Step>
+</Steps>
+
+<Check>Sequin will now capture changes from your Supabase database.</Check>
```
Comment (acco) [inline docs/guides/supabase.mdx:65]:
```suggestion
<Check>Sequin will now capture rows and changes from your Supabase database.</Check>
```

File: docs/guides/supabase.mdx [hunk 2]
```diff
@@ -1,7 +1,258 @@
...
+
+## Moving Supabase webhooks to Sequin
+
+Sequin adds some helpful features to Supabase webhooks:
```
Comment (acco) [inline docs/guides/supabase.mdx:69]:
```suggestion
Sequin has some helpful features that Supabase webhooks lack:

```

File: docs/guides/supabase.mdx [hunk 3]
```diff
@@ -1,7 +1,258 @@
...
+* Exactly-once processing guarantees ensuring every event is processed once and only once.
+* Automatic retries and backoff for failed events.
+* Filtering of events with no plpgsql.
```
Comment (acco) [inline docs/guides/supabase.mdx:72]:
```suggestion
* Filtering of events with no PL/pgSQL.
```

File: docs/guides/supabase.mdx [hunk 4]
```diff
@@ -1,7 +1,258 @@
...
+* Backfill support to sync historical data.
+* Helpful observability and debugging tools.
+
+You can easily move your existing Supabase webhooks to Sequin by creating Sequin HTTP push consumers. Here is a quick example of how to do this using a hypothetical `orders` table:
```
Comment (acco) [inline docs/guides/supabase.mdx:76]: HTTP push consumers  -> Link to docs section

```suggestion
You can easily move your existing Supabase webhooks to Sequin by creating Sequin HTTP push consumers. Here is an example of how to do this using a hypothetical `orders` table:
```

File: docs/guides/supabase.mdx [hunk 5]
```diff
@@ -1,7 +1,258 @@
...
+
+### Supabase webhook
```
Comment (acco) [inline docs/guides/supabase.mdx:78]: This section header confused me

File: docs/guides/supabase.mdx [hunk 6]
```diff
@@ -1,7 +1,258 @@
...
+
+Everytime an new order is created, a postgres trigger fires a webhook to send the data to the `orders` endpoint:
```
Comment (acco) [inline docs/guides/supabase.mdx:80]:
```suggestion
Everytime a new order is created, a Supabase Postgres trigger fires a webhook to send the data to the `orders` endpoint:
```

Comment (acco) [reply docs/guides/supabase.mdx:80]: Still confused, why is this sentence ending with a `:`?

File: docs/guides/supabase.mdx [hunk 7]
```diff
@@ -1,7 +1,258 @@
...
+
+Here is the configuration for the webhook in the Supabase:
```
Comment (acco) [inline docs/guides/supabase.mdx:82]: grammar

File: docs/guides/supabase.mdx [hunk 8]
```diff
@@ -1,7 +1,258 @@
...
+    ```
+  </Tab>
+  <Tab title="Supabase Console">
+    <Frame>
+        <img src="../images/guides/supabase/supabase-webhook.png" alt="Supabase webhook" />
+    </Frame>
+  </Tab>
+</Tabs>
+
+When this trigger fires, the data is sent to the `orders` endpoint with the following payload:
```
Comment (acco) [inline docs/guides/supabase.mdx:100]: use active voice, not passive

File: docs/guides/supabase.mdx (hunk 5 above)
Comment (acco) [reply docs/guides/supabase.mdx:78]: Ah, looks like you're contrasting the two - I think that could be way clearer

File: docs/guides/supabase.mdx [hunk 9]
```diff
@@ -1,7 +1,258 @@
...
+  <Step title="Create a new push consumer">
+    In the Sequin Console, navigate to the Consumers page, click **Create Consumer**.
+
+    In the setup wizard, select **Change stream** as the stream type and then **HTTP Push** as the consumer type.
+  </Step>
+  <Step title="Select your database and table">
+    Select your Supabase database and the `public.orders` table.
+  </Step>
+  <Step title="Set filters and operations">
+    Don't configure any filters and set the consumer to capture just the `INSERT` operations:
```
Comment (acco) [inline docs/guides/supabase.mdx:131]:
```suggestion
    Don't configure any filters and set the consumer to capture just `INSERT` operations:
```

File: docs/guides/supabase.mdx [hunk 10]
```diff
@@ -1,7 +1,258 @@
...
+        <img src="../images/guides/supabase/sequin-consumer-filters.png" alt="Sequin consumer settings" />
+    </Frame>
+  </Step>
+  <Step title="Add your webhook URL">
+    Finally, give your consumer a name, set a timeout, and add your webhook URL:
+    <Frame>
+        <img src="../images/guides/supabase/sequin-consumer-endpoint.png" alt="Sequin consumer settings" />
+    </Frame>
+
+    <Note>Sequin gives you more control over webhook delivery and retries including a higher request timeout, automatic retries, and the ability to configure max ack pending (e.g. the consumer will stop sending new events to the endpoint if a certain number of events are not acknowledged). For even more control over event consumption, you can use Sequin's HTTP pull consumer.</Note>
```
Comment (acco) [inline docs/guides/supabase.mdx:142]: Not sure the sell to the pull consumer is needed

File: docs/guides/supabase.mdx [hunk 11]
```diff
@@ -1,7 +1,258 @@
...
+  "action": "insert",
+  "changes": null
+}
+```
+
+<Note>
+  Sequin's data payload is slightly different to Supabase's webhook payload:
+  * Sequin's `metadata` field includes additional metadata about the consumer and the table.
+  * Sequin captures the operation type (e.g. `insert`, `update`, `delete`) in the `action` field not the `type` field.
+  * Sequin captures changes for `update` operations in the `changes` field not the `old_record` field.
```
Comment (acco) [inline docs/guides/supabase.mdx:174]: I assume `changes` is also different than `old_record`? Do you need to explain `changes` here?

File: docs/guides/supabase.mdx [hunk 12]
```diff
@@ -1,7 +1,258 @@
...
+</Note>
+
+## Triggering a Supabase edge function with Sequin
+
+Sequin can also trigger Supabase edge functions by calling the function's endpoint URL. This is an alternative to triggering Supabase edge functions with a Supabase webhooks, but with some nice guarantees and observability. Notably, Sequin can retry failed events.
```
Comment (acco) [inline docs/guides/supabase.mdx:179]:
```suggestion
Sequin can trigger Supabase edge functions by calling the function's endpoint URL. This is an alternative to triggering Supabase edge functions with Supabase webhooks. Sequin improves on the Supabase workflow by offering more features, like retries for failed events.
```

File: docs/guides/supabase.mdx [hunk 13]
```diff
@@ -1,7 +1,258 @@
...
+
+<Steps>
+  <Step title="Update the edge function to process Sequin's payload">
+    Update the edge function to process Sequin's event payload. Note that while Sequin's record payload is the same, the changes, actions, and metadata are different:
```
Comment (acco) [inline docs/guides/supabase.mdx:183]: Interesting that this section is about _migrating_ vs adopting. I think that's wise, but could be clearer that this is what this section (and whole guide?) is about

Comment (acco) [reply docs/guides/supabase.mdx:183]:
```suggestion
    Update the edge function to process Sequin's event payload. Note that while Sequin's `record` payload is the same, the keys `changes`, `actions`, and `metadata` are different:
```

File: docs/guides/supabase.mdx [hunk 14]
```diff
@@ -1,7 +1,258 @@
...
+        "changes": null
+        }
+        ```
+    </CodeGroup>
+  </Step>
+  <Step title="Deploy the edge function and retrieve the URL">
+    Using the Supabase CLI, deploy your edge function:
+
+    ```bash
+    supabase functions deploy {your-function-name} --no-verify-jwt
```
Comment (acco) [inline docs/guides/supabase.mdx:228]: Is there any security on this endpoint at all? Any way to add security? That is a requirement.

File: docs/guides/supabase.mdx [hunk 15]
```diff
@@ -1,7 +1,258 @@
...
+  </Step>
+</Steps>
+
+<Check>Now, when you make a change in your Supabase database that meets the consumer's filter criteria, Sequin will capture the change and send it to the edge function.</Check>
+
+## Next steps
+
+Sequin is a great complement to Supabase. This guide showed you how to connect Supabase to Sequin and start moving your webhooks and edge function triggers to Sequin. From here, you may want to:
+
+* Refine your webhooks / triggers by using more specific filters. Perhaps you can move logic out of your edge function and into your consumer filters.
```
Comment (acco) [inline docs/guides/supabase.mdx:255]: Nice

File: docs/guides/supabase.mdx [hunk 16]
```diff
@@ -1,7 +1,258 @@
...
+* Use Sequin's HTTP pull consumer to gain more control over event consumption.
```
Comment (acco) [inline docs/guides/supabase.mdx:256]: Link

**Comment by thisisgoldman** [comment]
### Merge activity

* **Sep 3, 11:19 PM EDT**: @thisisgoldman started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/sequinstream/sequin/203).

**Pull request merged**